libs/blocks/preflight/panels/assets.js

```diff
@@ -0,0 +1,188 @@
+import { html, signal, useEffect } from '../../../deps/htm-preact.js';
+import { createTag } from '../../../utils/utils.js';
+
+function isViewportTooSmall() {
+  return !window.matchMedia('(min-width: 1200px)').matches;
+}
+
+const imagesWithMismatch = signal([]);
+const imagesWithMatch = signal([]);
+const checksPerformed = signal(false);
+const viewportTooSmall = signal(isViewportTooSmall());
+
+const groups = [
+  { title: 'Images with dimension mismatch', imgArray: imagesWithMismatch },
+  { title: 'Images with matching dimensions', imgArray: imagesWithMatch },
+];
+
+async function checkImageDimensions() {
+  if (checksPerformed.value || viewportTooSmall.value) return;
+  checksPerformed.value = true;
+
+  const allImages = [...document.querySelectorAll('main picture img')];
```

Comment:
I’ve noticed that headers sometimes include images too. It might be a good idea to check those as well. What’s your take on that?

Reply:
Right now the big pain points revolve around marquee and aside images, which are also the ones that are instantly visible to the user without performing any additional action. Moreover, the browser doesn't have good layout data for hidden areas, so we'd have to force display any such areas in order to perform calculations. For the MVP, I think it's ok to ignore the `header` section.

Reply:
Sounds good. If this is going to get built out then , then we'll probably also want to some logic for images hidden behind tabs and other such cases.

Reply:
Yes, that's a great point! I'll create a due diligence to iterate over this solution and also include this caveat that we'll need to consider hidden section as well in the future or at least surface hidden images that could not be analyzed.